glue/app/qt/layer_tree_widget.py

```diff
@@ -197,14 +198,14 @@ class DeleteAction(LayerAction):
 
     def _can_trigger(self):
         selection = self.selected_layers()
-        return all(isinstance(s, (core.Data, core.SubsetGroup))
+        return all(isinstance(s, (core.Data, core.SubsetGroup, IndexedData, SlicedData))
```

Comment:
I think here you should just replace Data, IndexedData and SlicedData by BaseData